updatehub/tests/common.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright (C) 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use mockito::{mock, Mock};
+use regex::Regex;
+use serde_json::json;
+use std::{env, path::PathBuf};
+
+pub enum FakeServer {
+    NoUpdate,
+    HasUpdate(String),
+}
+
+pub enum Polling {
+    Enable,
+    Disable,
+}
+
+pub enum Server {
+    Custom(String),
+    Standard,
+}
+
+pub struct Settings {
+    polling: bool,
+    listen_socket: String,
+    server_address: String,
+    download_dir: Option<PathBuf>,
+    config_file: Option<PathBuf>,
+    timeout: Option<u64>,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Settings {
+            polling: false,
+            listen_socket: String::from("localhost:8080"),
+            server_address: mockito::server_url(),
+            download_dir: None,
+            config_file: None,
+            timeout: None,
+        }
+    }
+}
+
+impl Settings {
+    pub fn init_server(self) -> (rexpect::session::PtySession, updatehub::tests::TestEnvironment) {
+        let setup = updatehub::tests::TestEnvironment::build()
+            .listen_socket(self.listen_socket)
+            .server_address(self.server_address)
+            .add_echo_binary("reboot");
+
+        let mut setup = if !self.polling { setup.disable_polling() } else { setup }.finish();
+
+        if let Some(download_dir) = self.download_dir {
+            setup.settings.data.update.download_dir = download_dir;
+
+            let content = toml::ser::to_string_pretty(&setup.settings.data.update.download_dir)
+                .expect("fail to convert the data to toml");
+            std::fs::write(&setup.settings.stored_path, content)
+                .expect("fail to write the content on settings file");
```

Comment:
Are we storing the `download_dir` field as it it was the hole `Settings` file? This doesn't seem right.

Reply:
I fixed right now this.